updates/src/process.js

```diff
@@ -20,30 +21,95 @@ function removedForVersion(version) {
 function mapStandaloneCompat(engines, { support, ...compat }) {
   return {
     ...compat,
-    engines: [
-      ...new Set(
-        Object.entries(support || {})
-          .map(([browser, s]) => {
-            const { version, engine } = engines[browser] || {};
-            const removed = s.find(({ version_removed }) =>
-              Boolean(version_removed)
-            );
-            const added = s.find(({ version_added }) => Boolean(version_added));
-            if (
-              added &&
-              gte(version, added.version_added) &&
-              !(removed && gte(version, removed.version_removed))
-            ) {
-              return engine;
-            }
-            return null;
-          })
-          .filter(Boolean)
-      ),
-    ],
+    engines: mapEngines(engines, { support }),
   };
 }
 
+function buildStableVersionAndEngineByDate(browsers) {
+  const lookup = new Map();
+  for (const [browser, { releases }] of Object.entries(browsers)) {
+    const all = Object.entries(releases)
+      .map(([version, release]) => {
+        return { version, ...release };
+      })
+      .filter(({ release_date }) => Boolean(release_date));
+    all.sort(({ release_date: a }, { release_date: b }) => {
+      if (a < b) return -1;
+      if (b < a) return 1;
+      return 0;
+    });
+    lookup.set(browser, all);
+  }
+  return lookup;
+}
+
+let STABLE_VERSION_AND_ENGINE_BY_DATE;
+
+function stableVersionAndEngineByDate(browser, release_date) {
+  const all = STABLE_VERSION_AND_ENGINE_BY_DATE.get(browser);
+  if (!all) {
+    return null;
+  }
+  const idx = all.findLastIndex(({ release_date: r }) => r <= release_date);
+  if (idx > -1) {
+    const { version, engine } = all[idx];
+    return { version, engine: engine?.toLowerCase() };
+  }
+  return null;
+}
+
+function engineSupport(release_date, { support }) {
```

Comment:
If I've understood correctly, I'm not sure this logic is quite right, for instance:

If Edge adds support for a feature which Chrome hasn't (through some set of patches on top of Blink or whatever), Blink will show as supporting a particular feature.

If Chrome removes a feature, it'll remain showing as supported in Blink until every Blink-based browser releases a stable version also removing it.

IMO we should maintain an "authoritative" browser for each engine - or at least certain ones.